src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.ParsedSyntaxTree.cs

```diff
@@ -121,6 +115,7 @@ public override CSharpParseOptions Options
                 }
             }
 
+            [Obsolete("Obsolete due to performance problems, use CompilationOptions.SyntaxTreeOptionsProvider instead", error: true)]
```

Comment:
These aren't source-breaking, should we consider making them warnings instead?

Reply:
So I had looked through the compiler and it seems there's a mix of both error: true and error: false, with no clear rationale to me as to why one was being picked over another. And it may not be a source breaking change in that your code won't compile, but this is a functional break in the sense of your code doesn't do what you wrote it to do; I'm not sure there's a meaningful difference there -- you're broken.

That said, I've switched it back to error: false only because we still have code calling _into_ these which is in non-obsolete methods, and it seems once it's error: true I can't suppress those in any way. Which makes sense in that you can't suppress any _other_ compile error, but at the same time the compiler could still emit code just fine in this case.